picard/config.py

```diff
@@ -129,6 +131,51 @@ def value(self, name, option_type, default=None):
         return default
 
 
+class SettingConfigSection(ConfigSection):
+    """Custom subclass to automatically accommodate saving and retrieving values based on user profile settings.
+    """
+    def __init__(self, config, name):
+        super().__init__(config, name)
+        self.__qt_config = config
+        self.__name = name
+        self.__prefix = self.__name + '/'
+        self._memoization = defaultdict(Memovar)
+        Option.add_if_missing(name, 'user_profiles', [])
+
+    def _get_profiles(self):
+        profiles = self['user_profiles']
+        return profiles if profiles is not None else []
```

Comment:
We could make this a generator returning only active profiles:

```python
def _get_active_profiles(self):
     profiles = self['user_profiles']
     if profiles:
           for profile in profiles:
               if profile['active']:
                   yield profile
```

It would help to simplify code below.

```python
for profile in self._get_active_profiles():
      value = getattr(profile['settings'], name, None)
      if value is not None:
           return value
```

Reply:
Good suggestion.  I also found a problem where `getattr` wasn't returning the value, so I modified that a bit as well.